skyportal/handlers/api/gcn.py

```diff
@@ -0,0 +1,202 @@
+# Inspired by https://github.com/growth-astro/growth-too-marshal/blob/main/growth/too/gcn.py
+
+import datetime
+import os
+import gcn
+import lxml
+import xmlschema
+from urllib.parse import urlparse
+
+from baselayer.app.env import load_env
+
+from sqlalchemy.orm.exc import NoResultFound
+
+from baselayer.app.access import auth_or_token
+from ..base import BaseHandler
+from ...models import (
+    DBSession,
+    GcnEvent,
+    GcnNotice,
+    GcnTag,
+    Localization,
+)
+from ...utils.gcn import get_dateobs, get_tags, get_skymap, get_contour
+
+_, cfg = load_env()
+
+
+class GcnHandler(BaseHandler):
+    @auth_or_token
+    def put(self):
+        """
+        ---
+        description: Ingest GCN xml file
```

Comment:
We should document here the required top-level JSON body contents (i.e. the `"xml"` parameter)

Reply:
I went through and added Schema's to schema.py... hopefully this was a sensible choice based on the other examples?

Reply:
Yep, looks great!

Reply:
Thanks Ari!